test/discr/lp_discr_test.py

```diff
@@ -273,6 +273,60 @@ def test_element_from_array_2d_shape():
                        [3, 4]])
 
 
+def test_element_from_function_1d():
+
+    space = odl.uniform_discr(-1, 1, 4)
+    points = space.grid.points().squeeze()
+
+    # Without parameter
+    def f(x):
+        return x * 2 + np.maximum(x, 0)
+
+    elem_f = space.element(f)
+    true_elem = [x * 2 + max(x, 0) for x in points]
+    assert all_equal(elem_f, true_elem)
+
+    # With parameter
+    def f(x, **kwargs):
```

Comment:
The problem here is that the second positional argument is interpreted as `out` and has to have the name `out` since we pass the function through `uspace.element`, which makes a `FunctionSpaceVector` from it, which in turn is based on `Operator`. That's where the restriction comes from.